insights/core/filters.py

```diff
@@ -81,18 +81,25 @@ def inner(component, patterns):
 
         for pat in patterns:
             if not pat:
-                raise Exception("Filter patterns must not be empy.")
+                raise Exception("Filter patterns must not be empty.")
 
-        FILTERS[component] |= patterns
+        FILTERS[comp] |= patterns
+
+    def get_dependency_datasources(comp):
+        """"Get (all) the first depended datasource"""
+        dep_ds = set()
+        if plugins.is_datasource(comp):
+            dep_ds.add(comp)
+            return dep_ds
+        for dep in dr.get_dependencies(comp):
+            dep_ds.update(get_dependency_datasources(dep))
+        return dep_ds
 
     if not plugins.is_datasource(component):
-        deps = dr.run_order(dr.get_dependency_graph(component))
+        deps = get_dependency_datasources(component)
```

Comment:
Does not running the results of get_dependency_datasources through dr.run_order effect this in any way?

Reply:
@ryan-blakley - I think the `get_dependency_graph` is the reason that caused the duplicated filters when applying `add_filter` to a Parser/Combiner. So made this change. See.

```
In [6]: [dr.get_simple_name(dep) for dep in dr.get_dependency_graph(PsAux)]
Out[6]: ['PsAux', 'ps_aux', 'ps_aux', 'HostContext']

In [7]: [dr.get_name(dep) for dep in dr.get_dependency_graph(PsAux)]
Out[7]:
['insights.parsers.ps.PsAux',
 'insights.specs.Specs.ps_aux',
 'insights.specs.default.DefaultSpecs.ps_aux',
 'insights.core.context.HostContext']
```

Reply:
Gotcha, I think the run_order part confused me, I didn't know if there was a specific run order that was needed. After checking further I see all it did was sort it.